### PR TITLE
FIX Replace deprecated Config stat() calls with get()

### LINKED ISSUE #45: Replace deprecated Config stat() calls with get()
E.g.: https://github.com/silverstripe/silverstripe-restfulserver/pull/42/files#diff-f596ffd70cab74ef21dd377a7e5add53R117 https://github.com/silverstripe/silverstripe-restfulserver/pull/42/files#diff-f596ffd70cab74ef21dd377a7e5add53R146

### PR DESCRIPTION
**Pull request by raissanorth**
Resolves #45 

Also makes minor changes to formatting to comply with psr-2, deletes unused files and imports.

### CODE REVIEW

File: src/RestfulServer.php [hunk 1]
```diff
@@ -493,7 +493,7 @@ protected function postHandler($className, $id, $relation)
                 return $this->notFound();
             }
 
-            if (!$obj->stat('allowed_actions') || !in_array($relation, $obj->stat('allowed_actions'))) {
+            if (!$obj->get('allowed_actions') || !in_array($relation, $obj->get('allowed_actions'))) {
```
Comment (NightJar) [inline src/RestfulServer.php]: This is calling e.g. `Page::get`, where it should be `Config::get` - that is to say $obj isn't a `Config` instance, as the code seems to expect. You'll need to update this to:

```diff
- $obj->get(
+ Config::inst()->get($className,
```

File: src/RestfulServer.php [hunk 2]
```diff
@@ -703,7 +705,8 @@ protected function getAllowedRelations($class, $member = null)
                 //remove dot notation from relation names
                 $parts = explode('.', $relClass);
                 $relClass = array_shift($parts);
-                if (singleton($relClass)->stat('api_access')) {
+//                if (singleton($relClass)->get('api_access')) {
```
Comment (NightJar) [inline src/RestfulServer.php]: Can remove the comment :)

File: src/RestfulServer.php [hunk 3]
```diff
@@ -297,7 +296,8 @@ protected function getSearchQuery(
         $sort = null,
         $limit = null,
         $existingQuery = null
-    ) {
+    )
+    {
```
Comment (NightJar) [inline src/RestfulServer.php]: I'm not sure this is PSR-2 compliant - you can read more at http://www.php-fig.org/psr/psr-2/
running `phpcs` with the psr-2 standard (use the `--standard` flag, and use the file in the module's root) should identify these things for you, making it easy :)

File: src/DataFormatter.php [hunk 4]
```diff
@@ -310,7 +311,7 @@ protected function getFieldsForObj($obj)
         }
 
         // add default required fields
-        $dbFields = array_merge($dbFields, array('ID'=>'Int'));
+        $dbFields = array_merge($dbFields, array('ID' => 'Int'));
```
Comment (NightJar) [inline src/DataFormatter.php]: Super low impact and not generally a thing to worry about eliminating from existence everywhere, but when we're updating code we should generally try to update to the new array syntax: `[` ... `]`

File: src/RestfulServer.php (hunk 3 above)
Comment (NightJar) [reply src/RestfulServer.php]: Yeah, see here :)
https://travis-ci.org/silverstripe/silverstripe-restfulserver/jobs/312211524#L639

File: src/RestfulServerList.php [hunk 5]
```diff
@@ -1,23 +0,0 @@
...
-    public function __construct($list)
-    {
-        $this->list = $list;
-    }
-
-    public function handleItem($request)
-    {
-        return new RestfulServerItem($this->list->getById($request->param('ID')));
-    }
-}
```
Comment (robbieaverill) [inline src/RestfulServerList.php]: Is there a reason we're removing these classes?

**Comment by robbieaverill** [comment]
Merge on green

**Pull request merged**